src/mrack/providers/aws.py

```diff
@@ -93,13 +93,15 @@ async def validate_hosts(self, hosts):
                 )
                 logger.error(err_msg)
                 err_resp = image_err.response["Error"]["Message"]
-                raise ValidationError(f"{err_msg} Request failed with {err_resp}")
+                raise ValidationError(
+                    f"{err_msg} Request failed with {err_resp}"
+                ) from image_err
 
         return
 
     async def can_provision(self, hosts):
         """Check that hosts can be provisioned."""
-        return True
+        return bool(hosts)
```

Comment:
This is bad. `can_provision([])` should return `True`.

Reply:
Is this really needed or expected? I think that provider without hosts should not even 'think' about provisioning.

Reply:
It's an edge case. "Can provision" is about resources and it would be weird to say no with 0 hosts and yes with e.g. 1.  To "think" about provisioning is  on somebody else.

Reply:
I believe that empty requirement is a corner case. Is my assumption correct if I say that this may happen only while using mrack as library.
AWS, Beaker, and Static for example they will always provision (well if Static provider got a empty list instead of existing resource it is a very bad) because there is different approach in openstack we have quota and in these we are 'billed' by instances. I believe this is not a big issue that we return false if this list is empty. 
I believe adding a `req` to the line with `raise ValidationError(f"{self.dsp_name}: Not enough resources to provision")` and having errors more distinguished might be good as well. For example:
```python
        if not await self.can_provision(hosts):
            if not hosts:
                raise ProvisioningError("Can not continue with empty requirement for provider")
            else:
                raise ValidationError(f"{self.dsp_name}: Not enough resources to provision")
        logger.info(f"{self.dsp_name}: Resource availability: OK")
```

Reply:
Short answer: I think `can_provision` should return True for empty lists, given that there's no other logic there. 

Regarding the `if not hosts -> raise Exception`: this is the kind of validation I would expect to find within that method, but empty list of hosts could print a warning message perhaps. This way mrack as library can handle this before calling `can_provision` and mrack as tool will "never" encounter this.

Reply:
Fixed